diff -Naur WeDoMore.orig/wedo/__init__.py WeDoMore/wedo/__init__.py
--- WeDoMore.orig/wedo/__init__.py	2016-03-23 10:46:04.755922041 +0100
+++ WeDoMore/wedo/__init__.py	2016-03-23 10:47:14.099921868 +0100
@@ -7,6 +7,7 @@
 
 import os
 import usb.core
+import usb.backend.libusb1
 import logging
 
 logger = logging.getLogger('wedo')
@@ -28,7 +29,8 @@
     """ Find all available devices """
     devices = []
     try:
-        for dev in usb.core.find(find_all=True, idVendor=ID_VENDOR, idProduct=ID_PRODUCT):
+        backend = usb.backend.libusb1.get_backend(find_library=lambda x: "/usr/lib/libusb-1.0.so")
+        for dev in usb.core.find(find_all=True, idVendor=ID_VENDOR, idProduct=ID_PRODUCT, backend=backend):
             devices.append(dev)
     except usb.core.USBError as e:
         logger.error("Could not find a connected WeDo device: %s" % str(e))
diff -Naur WeDoMore.orig/wedo_plugin/wedo/__init__.py WeDoMore/wedo_plugin/wedo/__init__.py
