--- psplash.orig/psplash.c     2021-10-29 14:44:47.000000000 +0200
+++ psplash-44afb7506d43cca15582b4c5b90ba5580344d75d/psplash.c 2023-04-22 00:36:49.918344557 +0200
@@ -20,6 +20,11 @@
 #endif
 #include FONT_HEADER
 
+int YPos = -1;
+unsigned long BarCol=0xFFFFFFFF;
+unsigned long BarBkCol=0xFFFFFFFF;
+unsigned long BkCol=0xFFFFFFFF;
+
 #define SPLIT_LINE_POS(fb)                                  \
 	(  (fb)->height                                     \
 	 - ((  PSPLASH_IMG_SPLIT_DENOMINATOR                \
@@ -46,13 +51,28 @@
 
   /* Clear */
 
+  if(BkCol == 0xFFFFFFFF)
   psplash_fb_draw_rect (fb, 
 			0, 
 			SPLIT_LINE_POS(fb) - h, 
 			fb->width,
 			h,
 			PSPLASH_BACKGROUND_COLOR);
+  else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BkCol & 0x000000FF);
+            g = (uint8)((BkCol >> 8) & 0x000000FF);
+            r = (uint8)((BkCol >> 16) & 0x000000FF);
 
+   psplash_fb_draw_rect (fb, 
+			0, 
+			SPLIT_LINE_POS(fb) - h, 
+			fb->width,
+			h,
+			r, g, b);
+  }  
+ 
   psplash_fb_draw_text (fb,
 			(fb->width-w)/2, 
 			SPLIT_LINE_POS(fb) - h,
@@ -69,28 +89,84 @@
 
   /* 4 pix border */
   x      = ((fb->width  - BAR_IMG_WIDTH)/2) + 4 ;
-  y      = SPLIT_LINE_POS(fb) + 4;
+  if(YPos < 0)
+    y      = SPLIT_LINE_POS(fb) + 4;
+  else
+    y      = YPos + 4;
+
   width  = BAR_IMG_WIDTH - 8; 
   height = BAR_IMG_HEIGHT - 8;
 
   if (value > 0)
     {
       barwidth = (CLAMP(value,0,100) * width) / 100;
-      psplash_fb_draw_rect (fb, x + barwidth, y, 
+      if(BarBkCol == 0xFFFFFFFF)
+       psplash_fb_draw_rect (fb, x + barwidth, y, 
     			width - barwidth, height,
 			PSPLASH_BAR_BACKGROUND_COLOR);
-      psplash_fb_draw_rect (fb, x, y, barwidth,
+      else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BarBkCol & 0x000000FF);
+            g = (uint8)((BarBkCol >> 8) & 0x000000FF);
+            r = (uint8)((BarBkCol >> 16) & 0x000000FF);
+       psplash_fb_draw_rect (fb, x + barwidth, y, 
+    			width - barwidth, height,
+			r, g, b);
+      }
+      
+      			
+			
+      if(BarCol == 0xFFFFFFFF)			
+       psplash_fb_draw_rect (fb, x, y, barwidth,
 			    height, PSPLASH_BAR_COLOR);
+       else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BarCol & 0x000000FF);
+            g = (uint8)((BarCol >> 8) & 0x000000FF);
+            r = (uint8)((BarCol >> 16) & 0x000000FF);
+       psplash_fb_draw_rect (fb, x, y, barwidth,
+			    height, r, g, b);
+            
+            
+       }
     }
   else
     {
       barwidth = (CLAMP(-value,0,100) * width) / 100;
-      psplash_fb_draw_rect (fb, x, y, 
+
+
+      if(BarBkCol == 0xFFFFFFFF)
+       psplash_fb_draw_rect (fb, x, y, 
     			width - barwidth, height,
 			PSPLASH_BAR_BACKGROUND_COLOR);
-      psplash_fb_draw_rect (fb, x + width - barwidth,
+      else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BarBkCol & 0x000000FF);
+            g = (uint8)((BarBkCol >> 8) & 0x000000FF);
+            r = (uint8)((BarBkCol >> 16) & 0x000000FF);
+       psplash_fb_draw_rect (fb, x, y, 
+    			width - barwidth, height,
+			r, g, b);
+      }
+
+
+      if(BarCol == 0xFFFFFFFF)
+       psplash_fb_draw_rect (fb, x + width - barwidth,
 			    y, barwidth, height,
 			    PSPLASH_BAR_COLOR);
+      else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BarCol & 0x000000FF);
+            g = (uint8)((BarCol >> 8) & 0x000000FF);
+            r = (uint8)((BarCol >> 16) & 0x000000FF);
+       psplash_fb_draw_rect (fb, x + width - barwidth,
+			    y, barwidth, height,
+			    r, g, b);
+      }			    
     }
 
   DBG("value: %i, width: %i, barwidth :%i\n", value, 
@@ -227,6 +303,9 @@
   PSplashFB *fb;
   bool       disable_console_switch = FALSE;
 
+  char 	  FileName[250];
+  memset(FileName, 0, sizeof(FileName));
+
   signal(SIGHUP, psplash_exit);
   signal(SIGINT, psplash_exit);
   signal(SIGQUIT, psplash_exit);
@@ -251,10 +330,46 @@
         fbdev_id = atoi(argv[i]);
         continue;
       }
+    if (!strcmp(argv[i],"-y") || !strcmp(argv[i],"--YPos"))
+      {
+        if (++i >= argc) goto fail;
+        YPos = atoi(argv[i]);
+        continue;
+      }
+
+    if (!strcmp(argv[i],"--BarCol"))
+      {
+        if (++i >= argc) goto fail;
+        sscanf(argv[i],"%lX", &BarCol);
+        continue;
+      }
+
+    if (!strcmp(argv[i],"--BarBkCol"))
+      {
+ 	    if (++i >= argc) goto fail;
+ 	    sscanf(argv[i],"%lX", &BarBkCol);
+ 	    continue;
+ 	  }
+
+    if (!strcmp(argv[i],"--BkCol"))
+      {
+        if (++i >= argc) goto fail;
+        sscanf(argv[i],"%lX", &BkCol);
+ 	    continue;
+ 	  }
+      
+    if (!strcmp(argv[i],"--filename"))
+      {
+        if (++i >= argc) goto fail;
+        sprintf(FileName, argv[i]);
+        angle = atoi(argv[i]);
+        continue;
+      }
 
     fail:
       fprintf(stderr, 
-              "Usage: %s [-n|--no-console-switch][-a|--angle <0|90|180|270>][-f|--fbdev <0..9>]\n", 
+              "Usage: %s [-n|--no-console-switch][-a|--angle <0|90|180|270>][-f|--fbdev <0..9>]\n"
+              "\t\t [-y|--YPos <pos>][--filename <filename>][--BarCol <col>][--BarBkCol <col>][--BkCol <col>]\n",
               argv[0]);
       exit(-1);
   }
@@ -297,11 +412,12 @@
 #endif
 
   /* Clear the background with #ecece1 */
-  psplash_fb_draw_rect (fb, 0, 0, fb->width, fb->height,
+  if(strlen(FileName) == 0) {
+    psplash_fb_draw_rect (fb, 0, 0, fb->width, fb->height,
                         PSPLASH_BACKGROUND_COLOR);
 
-  /* Draw the Poky logo  */
-  psplash_fb_draw_image (fb, 
+    /* Draw the Poky logo  */
+    psplash_fb_draw_image (fb, 
 			 (fb->width  - POKY_IMG_WIDTH)/2, 
 #if PSPLASH_IMG_FULLSCREEN
 			 (fb->height - POKY_IMG_HEIGHT)/2,
@@ -314,12 +430,19 @@
 			 POKY_IMG_BYTES_PER_PIXEL,
 			 POKY_IMG_ROWSTRIDE,
 			 POKY_IMG_RLE_PIXEL_DATA);
+  } else {
+       char Cmd[512];
+       memset(Cmd, 0, sizeof(Cmd));
+       sprintf(Cmd,"fbv -s1 %s", FileName);
+       system(Cmd); /* Draw file with fbv command */
+  }
 
 #ifdef PSPLASH_SHOW_PROGRESS_BAR
   /* Draw progress bar border */
+  if ( BkCol == 0xFFFFFFFF)
   psplash_fb_draw_image (fb, 
 			 (fb->width  - BAR_IMG_WIDTH)/2, 
-			 SPLIT_LINE_POS(fb),
+			 (YPos < 0)?SPLIT_LINE_POS(fb):YPos,
 			 BAR_IMG_WIDTH,
 			 BAR_IMG_HEIGHT,
 			 BAR_IMG_BYTES_PER_PIXEL,
@@ -330,7 +453,7 @@
 #endif
 
 #ifdef PSPLASH_STARTUP_MSG
-  psplash_draw_msg (fb, PSPLASH_STARTUP_MSG);
+  if(strlen(FileName) == 0) psplash_draw_msg (fb, PSPLASH_STARTUP_MSG);
 #endif
 
   /* Scene set so let's flip the buffers. */
