--- python-pillow-c01f51dfff07416aba5f9bd78717be1887f67868.orig/setup.py	2016-04-07 09:32:10.162132286 +0200
+++ python-pillow-c01f51dfff07416aba5f9bd78717be1887f67868/setup.py	2016-04-07 09:46:32.868701038 +0200
@@ -235,7 +235,8 @@
                 _add_directory(library_dirs, "/usr/X11/lib")
                 _add_directory(include_dirs, "/usr/X11/include")
 
-        elif sys.platform.startswith("linux"):
+#TH        elif sys.platform.startswith("linux"):
+        elif False:
             arch_tp = (plat.processor(), plat.architecture()[0])
             if arch_tp == ("x86_64", "32bit"):
                 # 32-bit build on 64-bit machine.
@@ -346,17 +347,17 @@
         # add standard directories
 
         # look for tcl specific subdirectory (e.g debian)
-        if _tkinter:
-            tcl_dir = "/usr/include/tcl" + TCL_VERSION
-            if os.path.isfile(os.path.join(tcl_dir, "tk.h")):
-                _add_directory(include_dirs, tcl_dir)
+#TH        if _tkinter:
+#TH            tcl_dir = "/usr/include/tcl" + TCL_VERSION
+#TH            if os.path.isfile(os.path.join(tcl_dir, "tk.h")):
+#TH                _add_directory(include_dirs, tcl_dir)
 
         # standard locations
-        _add_directory(library_dirs, "/usr/local/lib")
-        _add_directory(include_dirs, "/usr/local/include")
+#TH        _add_directory(library_dirs, "/usr/local/lib")
+#TH        _add_directory(include_dirs, "/usr/local/include")
 
-        _add_directory(library_dirs, "/usr/lib")
-        _add_directory(include_dirs, "/usr/include")
+#TH        _add_directory(library_dirs, "/usr/lib")
+#TH        _add_directory(include_dirs, "/usr/include")
 
         # on Windows, look for the OpenJPEG libraries in the location that
         # the official installer puts them
@@ -718,10 +719,10 @@
                 fp = open(tmpfile, 'r')
                 multiarch_path_component = fp.readline().strip()
                 fp.close()
-                _add_directory(self.compiler.library_dirs,
-                               '/usr/lib/' + multiarch_path_component)
-                _add_directory(self.compiler.include_dirs,
-                               '/usr/include/' + multiarch_path_component)
+#TH                _add_directory(self.compiler.library_dirs,
+#TH                               '/usr/lib/' + multiarch_path_component)
+#TH                _add_directory(self.compiler.include_dirs,
+#TH                               '/usr/include/' + multiarch_path_component)
         finally:
             os.unlink(tmpfile)
 
