From 1401e4ab9d1844fdf104a6af465ced1d5af4295c Mon Sep 17 00:00:00 2001
From: Raphael Jacob <r.jacob2002@gmail.com>
Date: Sat, 31 Dec 2022 00:24:45 +0100
Subject: [PATCH 1/1] use python3

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 581d7f5..2664f10 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/cmake_uninstall.cmake.in" "${CMAKE_C
 ADD_CUSTOM_TARGET(uninstall "${CMAKE_COMMAND}" -P "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake")
 
 # python & diag
-FIND_PROGRAM(PYTHON_EXECUTABLE python2)
+FIND_PROGRAM(PYTHON_EXECUTABLE python3)
 IF (PYTHON_EXECUTABLE)
 	FIND_PROGRAM(INSTALL_EXECUTABLE install)
 	MESSAGE(STATUS "Found Python: " ${PYTHON_EXECUTABLE})
-- 
2.30.2

