--- psplash.orig/psplash.c	2016-03-02 11:52:31.088994247 +0100
+++ psplash-14c8f7b705de944beb4de3f296506d80871e410f/psplash.c	2016-03-02 11:54:22.488993969 +0100
@@ -27,6 +27,8 @@
 
 int YPos = -1;
 unsigned long BarCol=0xFFFFFFFF;
+unsigned long BarBkCol=0xFFFFFFFF;
+unsigned long BkCol=0xFFFFFFFF;
 
 #define SPLIT_LINE_POS(fb)                                  \
 	(  (fb)->height                                     \
@@ -54,13 +56,28 @@
 
   /* Clear */
 
-  psplash_fb_draw_rect (fb, 
+  if(BkCol == 0xFFFFFFFF)
+   psplash_fb_draw_rect (fb, 
 			0, 
 			SPLIT_LINE_POS(fb) - h, 
 			fb->width,
 			h,
 			PSPLASH_BACKGROUND_COLOR);
+  else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BkCol & 0x000000FF);
+            g = (uint8)((BkCol >> 8) & 0x000000FF);
+            r = (uint8)((BkCol >> 16) & 0x000000FF);
 
+   psplash_fb_draw_rect (fb, 
+			0, 
+			SPLIT_LINE_POS(fb) - h, 
+			fb->width,
+			h,
+			r, g, b);
+  }  
+ 
   psplash_fb_draw_text (fb,
 			(fb->width-w)/2, 
 			SPLIT_LINE_POS(fb) - h,
@@ -87,9 +104,22 @@
   if (value > 0)
     {
       barwidth = (CLAMP(value,0,100) * width) / 100;
-      psplash_fb_draw_rect (fb, x + barwidth, y, 
+      if(BarBkCol == 0xFFFFFFFF)
+       psplash_fb_draw_rect (fb, x + barwidth, y, 
     			width - barwidth, height,
 			PSPLASH_BAR_BACKGROUND_COLOR);
+      else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BarBkCol & 0x000000FF);
+            g = (uint8)((BarBkCol >> 8) & 0x000000FF);
+            r = (uint8)((BarBkCol >> 16) & 0x000000FF);
+       psplash_fb_draw_rect (fb, x + barwidth, y, 
+    			width - barwidth, height,
+			r, g, b);
+      }
+      
+      			
 			
       if(BarCol == 0xFFFFFFFF)			
        psplash_fb_draw_rect (fb, x, y, barwidth,
@@ -97,9 +127,9 @@
        else {
             uint8 r,g,b;
             
-            b = BarCol & 0x000000FF;
-            g = (BarCol >> 8) & 0x000000FF;
-            r = (BarCol >> 16) & 0x000000FF;
+            b = (uint8)(BarCol & 0x000000FF);
+            g = (uint8)((BarCol >> 8) & 0x000000FF);
+            r = (uint8)((BarCol >> 16) & 0x000000FF);
        psplash_fb_draw_rect (fb, x, y, barwidth,
 			    height, r, g, b);
             
@@ -109,12 +139,38 @@
   else
     {
       barwidth = (CLAMP(-value,0,100) * width) / 100;
-      psplash_fb_draw_rect (fb, x, y, 
+
+
+      if(BarBkCol == 0xFFFFFFFF)
+       psplash_fb_draw_rect (fb, x, y, 
     			width - barwidth, height,
 			PSPLASH_BAR_BACKGROUND_COLOR);
-      psplash_fb_draw_rect (fb, x + width - barwidth,
+      else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BarBkCol & 0x000000FF);
+            g = (uint8)((BarBkCol >> 8) & 0x000000FF);
+            r = (uint8)((BarBkCol >> 16) & 0x000000FF);
+       psplash_fb_draw_rect (fb, x, y, 
+    			width - barwidth, height,
+			r, g, b);
+      }
+
+
+      if(BarCol == 0xFFFFFFFF)
+       psplash_fb_draw_rect (fb, x + width - barwidth,
 			    y, barwidth, height,
 			    PSPLASH_BAR_COLOR);
+      else {
+            uint8 r,g,b;
+            
+            b = (uint8)(BarCol & 0x000000FF);
+            g = (uint8)((BarCol >> 8) & 0x000000FF);
+            r = (uint8)((BarCol >> 16) & 0x000000FF);
+       psplash_fb_draw_rect (fb, x + width - barwidth,
+			    y, barwidth, height,
+			    r, g, b);
+      }			    
     }
 
   DBG("value: %i, width: %i, barwidth :%i\n", value, 
@@ -265,7 +321,21 @@
       if (!strcmp(argv[i],"--BarCol"))
         {
 	  if (++i >= argc) goto fail;
-	  sscanf(argv[i],"%X", &BarCol);
+	  sscanf(argv[i],"%lX", &BarCol);
+	  continue;
+	}
+
+      if (!strcmp(argv[i],"--BarBkCol"))
+        {
+	  if (++i >= argc) goto fail;
+	  sscanf(argv[i],"%lX", &BarBkCol);
+	  continue;
+	}
+
+      if (!strcmp(argv[i],"--BkCol"))
+        {
+	  if (++i >= argc) goto fail;
+	  sscanf(argv[i],"%lX", &BkCol);
 	  continue;
 	}
       
@@ -281,6 +351,8 @@
       fprintf(stderr, 
 	      "Usage: %s [-n|--no-console-switch][-a|--angle <0|90|180|270>] [-f|--filename] [-y|--YPos]\n"
 	      "\t\t [--BarCol]\n"
+	      "\t\t [--BarBkCol]\n"
+	      "\t\t [--BkCol]\n"
 	      , 
 	      argv[0]);
       exit(-1);
@@ -345,7 +417,8 @@
   }
 
   /* Draw progress bar border */
-  psplash_fb_draw_image (fb, 
+  if ( BkCol == 0xFFFFFFFF)
+   psplash_fb_draw_image (fb, 
 			 (fb->width  - BAR_IMG_WIDTH)/2, 
 			 (YPos < 0)?SPLIT_LINE_POS(fb):YPos, 
 			 BAR_IMG_WIDTH,
@@ -354,6 +427,7 @@
 			 BAR_IMG_ROWSTRIDE,
 			 BAR_IMG_RLE_PIXEL_DATA);
 
+ 
   psplash_draw_progress (fb, 0);
 
 #ifdef PSPLASH_STARTUP_MSG
